_studio/shared/src/libmfx_core_vaapi.cpp

```diff
@@ -612,7 +612,9 @@ VAAPIVideoCORE::AllocFrames(
         } else
         {
             // external allocator
-            if (m_bSetExtFrameAlloc && request->Info.FourCC != MFX_FOURCC_P8)
+            if (m_bSetExtFrameAlloc &&
+                request->Info.FourCC != MFX_FOURCC_P8 &&
+                !(request->Type & MFX_MEMTYPE_INTERNAL_FRAME))
```

Comment:
I didn't get it, Anton. at line 617/619, which allocator we call to? seems to be external? If we change a behavior and now allocates internal surfaces via ext allocator, then why do we need this call?

Reply:
@dmitryermilov this code path still uses for alloc external resources.

Reply:
From my understanding, this code block is used only for one case: when decoder at Init needs to access all memId's which had been allocated by app. I'd suggest to add a comment here:
`
// make 'fake' Alloc call to retrieve memId's of surfaces already allocated by app.
sts = (*m_FrameAllocator.frameAllocator.Alloc)(m_FrameAllocator.frameAllocator.pthis, &temp_request, response);
`

Reply:
when please see a comment below